File: Algorithm/Leetcode/0000_0100/0011_0020/0019.cpp
// 414. Third Maximum Number

class Solution {
public:
    int thirdMax(vector<int>& nums) {
		std::set<int> res;
		for (auto n : nums)
		{
			res.insert(n);
		}
		auto back_iter = res.end();
		back_iter--;

		if (res.size() > 2)
		{
			back_iter--;
			back_iter--;		
		}
			
		return *back_iter;
		
    }
};

File: Algorithm/Leetcode/0000_0100/0011_0020/0019.py
# 414. Third Maximum Number

"""https://leetcode.com/problems/third-maximum-number/discuss/90201/A-python-amusing-solution-which-actually-beats-98..."""

class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        nums = set(nums)
        if len(nums) < 3:
            return max(nums)
        nums.remove(max(nums))
        nums.remove(max(nums))
        return max(nums)